### PR TITLE
fix: cell height on container or theme change

### PR DESCRIPTION
**Pull request by cbt1**

### CODE REVIEW

**Review by haxxmaxx — APPROVED** [review]
code lgtm

File: src/table/virtualized-table/Header.tsx [hunk 1]
```diff
@@ -7,10 +7,10 @@ import useResetHeader from './hooks/use-reset-header';
 
 const Header = (props: HeaderProps) => {
   const { rect, forwardRef, columns, pageInfo, headerStyle, rowHeight, columResizeHandler } = props;
-  const { layout } = useContextSelector(TableContext, (value) => value.baseProps);
+  const { layout, theme } = useContextSelector(TableContext, (value) => value.baseProps);
   const columnWidths = useContextSelector(TableContext, (value) => value.columnWidths);
 
-  useResetHeader(forwardRef, layout, pageInfo, columnWidths);
+  useResetHeader(forwardRef, layout, pageInfo, columnWidths, theme);
```
Comment (a-m-dev) [inline src/table/virtualized-table/Header.tsx:13]: why not directly passing `theme.name()` from here?

File: src/table/virtualized-table/Totals.tsx [hunk 2]
```diff
@@ -7,10 +7,10 @@ import useResetHeader from './hooks/use-reset-header';
 
 const Totals = (props: TotalsProps) => {
   const { rect, forwardRef, pageInfo, totals, rowHeight, columns } = props;
-  const { layout, styling } = useContextSelector(TableContext, (value) => value.baseProps);
+  const { layout, styling, theme } = useContextSelector(TableContext, (value) => value.baseProps);
   const columnWidths = useContextSelector(TableContext, (value) => value.columnWidths);
 
-  useResetHeader(forwardRef, layout, pageInfo, columnWidths);
+  useResetHeader(forwardRef, layout, pageInfo, columnWidths, theme);
```
Comment (a-m-dev) [inline src/table/virtualized-table/Totals.tsx:13]: here too

File: src/table/virtualized-table/hooks/use-heights.ts [hunk 3]
```diff
@@ -41,6 +42,14 @@ const useHeights = ({ columns, columnWidths, pageInfo, headerRef, totalsRef, tot
     boldText: true,
   });
 
+  const resizeAllHeaderCells = headerHeight.resizeAllCells;
+  const resizeAllTotalCells = totalsHeight.resizeAllCells;
+
+  const resizeCells = useCallback(() => {
+    resizeAllHeaderCells();
+    resizeAllTotalCells();
+  }, [resizeAllHeaderCells, resizeAllTotalCells]);
```
Comment (a-m-dev) [inline src/table/virtualized-table/hooks/use-heights.ts:51]: These dependencies are functions, right? 
My questions is, why this useCallback should depend on another useCallback(ed) function, which does not return anything in result, just changes some internal states of `use-dynamic-row-height`?

If those functions are already in memoized in their own useCallback, what is the point of using useCallback here once more? and is using useCallback here, affects the performance?

File: src/table/virtualized-table/Header.tsx (hunk 1 above)
Comment (cbt1) [reply src/table/virtualized-table/Header.tsx:13]: Will fix 👍

File: src/table/virtualized-table/hooks/use-heights.ts (hunk 3 above)
Comment (cbt1) [reply src/table/virtualized-table/hooks/use-heights.ts:51]: Not that much point in it to be honest. If I don't wrap them in a useCallback, I would have to return both functions from this hook.

But thinking of it now. Skipping the useCallback, might actually make it a bit more clear  what going in the consuming component, as I would be "force" to change naming a bit.

**Review by a-m-dev — APPROVED** [review]

**Pull request merged**